.github/workflows/main.yml

```diff
@@ -0,0 +1,78 @@
+# **what?** - This workflow contains unit tests for supported OS and
+# python versions as well as static code checks. This workflow
+# should require no secrets.
+# **why?** - To ensure all code being merged into the project passes a
+# minimum threshold of quality.
+# **when?** - This workflow will run on ever push to a protected
+# branch, for all pull requests, and when manually triggered.
+
+name: CI
+
+on:
+  push:
+    branches:
+      - "main"
+      - "develop"
+      - "*.latest"
+      - "releases/*"
+  pull_request:
+  workflow_dispatch:
+
+concurrency:
+  group: ci-${{ github.event_name }}-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  test:
+    name: ${{ matrix.tox-env }} ${{ matrix.python-version }} / ${{ matrix.os }}
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ubuntu-latest]
+        tox-env: [unit, mypy]
+        python-version: [3.6, 3.7, 3.8] # TODO: support unit testing for python 3.9
+        include:
+          - os: ubuntu-latest
+            tox-env: flake8
+            python-version: 3.8
+          - os: ubuntu-latest
+            tox-env: mypy
+            python-version: 3.6
+          - os: ubuntu-latest
+            tox-env: mypy
+            python-version: 3.7
+          - os: ubuntu-latest
+            tox-env: mypy
+            python-version: 3.8
+          - os: ubuntu-latest
+            tox-env: mypy
+            python-version: 3.9
```

Comment:
You don't have `3.9` in the above list so does this work?

Reply:
Yes, it does. The items in the `include` list are added after the matrix if evaluated and flattened I believe. I went back and forth this specific matrix. I think it is fine to just run flake8 and mypy once, with a single python version. I'll update it so that is the case.